### PR TITLE
Add kotlin sample

### PR DESCRIPTION
**Pull request by Tapchicoma**
#30: Reorganized project to have plugin itself in a separate directory and add samples folder. Currently only kotlin sample added.

Samples uses composite build to apply ktlint-gradle plugin build, so travis will check if changes in plugin doesn't break anything.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,14 @@
 language: java
 
 script:
-  - ./gradlew build --no-daemon
-  - ./gradlew ktlintCheck --no-daemon
+  - .plugin/gradlew -p ./plugin build --no-daemon
+  - .plugin/gradlew -p ./plugin ktlintCheck --no-daemon
+  - .samples/gradlew -p ./samples :kotlin:ktlintCheck --no-daemon
```
Comment (JLLeitschuh) [inline .travis.yml]: I think travis automatically adds `--no-daemon` to all gradle invocations by modifying `~/.gradle/gradle.properties`. But you can keep this if you want.

**Comment by JLLeitschuh** [comment]
Dang!!! This is pretty awesome. I was thinking about doing this, but you went and did it!!! Great work!!! I'm going to have to check out this branch locally before I hit the merge button.

Let me know when it's in a state you are happy with it and I'll check it out.

Great work!!! 😄

**Comment by Tapchicoma** [comment]
Better to merge this PR after #32 will be merged. Also then I will add android sample.

**Comment by JLLeitschuh** [comment]
#32 has merge conflicts that need to be resolved.

File: .gitignore [hunk 2]
```diff
@@ -142,7 +143,7 @@ tags
 
 ### Gradle ###
 .gradle
-/build/
+build/
```
Comment (JLLeitschuh) [inline .gitignore]: 👍

**Pull request merged**